colour/models/rgb/transfer_functions/filmlight_tlog.py

```diff
@@ -0,0 +1,216 @@
+
+__author__ = 'Colour Developers'
+__copyright__ = 'Copyright (C) 2013-2018 - Colour Developers'
+__license__ = 'New BSD License - http://opensource.org/licenses/BSD-3-Clause'
+__maintainer__ = 'Colour Developers'
+__email__ = 'colour-science@googlegroups.com'
+__status__ = 'Production'
+
+__all__ = ['log_encoding_FilmLight_T_Log', 'log_decoding_FilmLight_T_Log']
+
+
+def log_encoding_FilmLight_T_Log(x, w=128.0, g=16.0, o=0.075):
+    """
+    Defines the *FilmLight T-Log* log encoding curve.
+
+    Parameters
+    ----------
+    x : numeric or array_like
+        Linear reflection data :math`x`.
+    w : numeric, optional
+        Value of :math:`x` for :math:`t = 1.0`.
+    g : numeric, optional
+        Gradient at :math:`x = 0.0`.
+    o : numeric, optional
+        Value of :math:`t` for :math:`x = 0.0`.
+
+    Returns
+    -------
+    numeric or ndarray
+        *FilmLight T-Log* encoded data :math:`t`.
+
+    References
+    ----------
+    :cite:`Siragusano2018a`
+
+    Notes
+    -----
+
+    +------------+-----------------------+---------------+
+    | **Domain** | **Scale - Reference** | **Scale - 1** |
+    +============+=======================+===============+
+    | ``x``      | [0, 1]                | [0, 1]        |
+    +------------+-----------------------+---------------+
+
+    +------------+-----------------------+---------------+
+    | **Range**  | **Scale - Reference** | **Scale - 1** |
+    +============+=======================+===============+
+    | ``t``      | [0, 1]                | [0, 1]        |
+    +------------+-----------------------+---------------+
+
+    -   The following is an excerpt from the FilmLight colour space file
+        `./etc/colourspaces/FilmLight_TLog_EGamut.flspace` which can be
+        obtained by installing the free Baselight for Nuke plugin::
+
+            T-Log, a cineon driven log tone-curve developed by FilmLight.
+            The colour space is designed to be used as *Working Colour Space*.
+
+            Version 10.0
+            This is similar to Cineon LogC function.
+
+            The formula is... 
+            y = A + B*log(x + C)
+            ...where x,y are the log and linear values.
+
+            A,B,C are constants calculated from...
+            w = x value for y = 1.0
+            g = the gradient at x=0
+            o = y value for x = 0.0
+
+            We do not have an exact solution but the
+            formula for b gives an approximation. The
+            gradient is not g, but should be within a 
+            few percent for most sensible values of (w*g).
+
+    Examples
+    --------
+    >>> log_encoding_FilmLight_T_Log(0.18)  # doctest: +ELLIPSIS
+    0.3965678...
+    """
+
+    b = 1.0 / (0.7107 + 1.2359 * np.log(w * g))
+    gs = g / (1.0 - o)
+    C = b / gs
+    a = 1.0 - b * np.log(w + C)
+    y0 = a + b * np.log(C)
+    s = (1.0 - o) / (1.0 - y0)
+    A = 1.0 + (a - 1.0) * s
+    B = b * s
+    G = gs * s
+
+    x = to_domain_1(x)
```

Comment:
We try to do input variable conversion and general massaging as the first step when entering the function, would it be possible to move `x = to_domain_1(x)` a line 107 please?